sources/Algo/src/Algo.cpp

```diff
@@ -162,6 +162,25 @@ LoadDefinitionAlgorithm::operator()(const NodePtr& node) {
     loads_.emplace_back(it->id, node->id);
   }
 }
+/////////////////////////////////////////////////////////////////
+
+ControllerInterfaceDefinitionAlgorithm::ControllerInterfaceDefinitionAlgorithm(std::vector<HvdcLineDefinition>& hvdcLines) : hvdcLines_(hvdcLines) {}
+
+void
+ControllerInterfaceDefinitionAlgorithm::operator()(const NodePtr& node) {
+  for (auto converter : node->converterInterfaces) {
+    auto hvdcLine = converter.hvdcLine;
+    HvdcLineDefinition::Position position;
+    if (converter.converterId == hvdcLine->converter1.converterId) {
+      position = HvdcLineDefinition::Position::FIRST_IN_MAIN_COMPONENT;
+    } else if (converter.converterId == hvdcLine->converter2.converterId) {
+      position = HvdcLineDefinition::Position::SECOND_IN_MAIN_COMPONENT;
```

Comment:
So if I understand well if both are in the main connex component we fail today?

Reply:
Yes

Reply:
@cossartque Are we sure we won't have failures on some testcases that we didn't have earlier, or we didn't use testcases with HVDC at all ?

Reply:
For the moment, if both terminals are in the main cc, we do nothing, and dynawo uses the basic cpp model for hvdc and it works fine (in terms of robustness).
In the future, we will apply the proper Modelica model to improve the results quality but I wanted to do it in a different PR to avoid too large PR.

Reply:
So for the cases with HVDC that worked before, nothing will change (for the moment)
For the cases with HVDC that didn't work because they have one bus in the main cc and the other in a minor cc, they should work now.
For the cases with HVDC entirely in a minor cc, nothing will change and they normally should work as well

Reply:
so if my understanding is correct we shouldn't fail if the hvdc is fully included in the main cc but just do nothing?

Reply:
Yes that's it

Reply:
then we need an update here

Reply:
@PerninGuillaume has it been updated ?

Reply:
Not yet, if you want to check the case of both converters in the main connex component, then I need to change a few things to be able to know when we are in this case.

Reply:
Done